testing/ltp/Makefile

```diff
@@ -103,9 +119,23 @@ CFLAGS       += -I$(LTP_UNPACK)
 CFLAGS       += -I$(TESTDIR)/include
 endif
 
+# Relax warning checks to avoid expected compile errors:
+CFLAGS += -Wno-strict-prototypes -Wno-return-type -Wno-format -Wno-uninitialized
+CFLAGS += -Wno-unused-variable -Wno-unused-function -Wno-unused-but-set-variable -Wno-unused-value
+
+# Should be removed if possible in the future
+CFLAGS += -Wno-incompatible-pointer-types -Wno-overflow -Wno-int-to-pointer-cast
+
+# This should be defined in unistd.h
+CFLAGS += -D_POSIX_TIMEOUTS=1
+
+# The tests use and #if instead of #ifdef for this
+CFLAGS += -D__linux__=0
```

Comment:
should we provide a patch to ltp project?

Reply:
Moved this to a patch that I will upstream.  There is only one place where it is `#if __linux__` instead of `#ifdef __linux__`